From: Matthew Leeds <matthew.leeds@endlessm.com>
Date: Thu, 15 Nov 2018 23:20:46 -0800
Subject: app: Fix a couple memory leaks

Closes: #2330
Approved by: matthiasclasen

(cherry picked from commit 7d35de18e46f1629c89d0f74d2ea0ef723814009)
---
 app/flatpak-builtins-utils.c | 6 ++++--
 common/flatpak-dir-private.h | 1 +
 common/flatpak-dir.c         | 2 +-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/app/flatpak-builtins-utils.c b/app/flatpak-builtins-utils.c
index ddfb139..65e4977 100644
--- a/app/flatpak-builtins-utils.c
+++ b/app/flatpak-builtins-utils.c
@@ -441,8 +441,9 @@ flatpak_resolve_matching_refs (const char   *remote_name,
 
   if (chosen == 0)
     {
+      const char *dir_name = flatpak_dir_get_name_cached (dir);
       g_print (_("Similar refs found for ‘%s’ in remote ‘%s’ (%s):\n"),
-               opt_search_ref, remote_name, flatpak_dir_get_name (dir));
+               opt_search_ref, remote_name, dir_name);
       for (i = 0; i < refs_len; i++)
         g_print ("%d) %s\n", i + 1, refs[i]);
       chosen = flatpak_number_prompt (TRUE, 0, refs_len, _("Which do you want to use (0 to abort)?"));
@@ -477,7 +478,8 @@ flatpak_resolve_matching_remotes (gboolean        disable_interaction,
       for (i = 0; i < remote_dir_pairs->len; i++)
         {
           RemoteDirPair *pair = g_ptr_array_index (remote_dir_pairs, i);
-          g_print ("%d) %s (%s)\n", i + 1, pair->remote_name, flatpak_dir_get_name (pair->dir));
+          const char *dir_name = flatpak_dir_get_name_cached (pair->dir);
+          g_print ("%d) %s (%s)\n", i + 1, pair->remote_name, dir_name);
         }
       chosen = flatpak_number_prompt (TRUE, 0, remote_dir_pairs->len, _("Which do you want to use (0 to abort)?"));
       if (chosen == 0)
diff --git a/common/flatpak-dir-private.h b/common/flatpak-dir-private.h
index 3cb2431..66e4052 100644
--- a/common/flatpak-dir-private.h
+++ b/common/flatpak-dir-private.h
@@ -275,6 +275,7 @@ GFile *     flatpak_dir_get_changed_path (FlatpakDir *self);
 const char *flatpak_dir_get_id (FlatpakDir *self);
 const char *flatpak_dir_get_display_name (FlatpakDir *self);
 char *      flatpak_dir_get_name (FlatpakDir *self);
+const char *flatpak_dir_get_name_cached (FlatpakDir *self);
 gint        flatpak_dir_get_priority (FlatpakDir *self);
 FlatpakDirStorageType flatpak_dir_get_storage_type (FlatpakDir *self);
 GFile *     flatpak_dir_get_deploy_dir (FlatpakDir *self,
diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index b5f2721..73a6705 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -1596,7 +1596,7 @@ flatpak_dir_get_name (FlatpakDir *self)
   return g_strdup ("system");
 }
 
-static const char *
+const char *
 flatpak_dir_get_name_cached (FlatpakDir *self)
 {
   char *name;
